python/ray/experimental/state.py

```diff
@@ -166,9 +166,7 @@ def _initialize_global_state(self,
 
             # If we got here then we successfully got all of the information.
             break
-
-        # Check to see if we timed out.
-        if time.time() - start_time >= timeout:
+        else:
```

Comment:
Hmm, `while: else:` is a pretty weird syntax. I agree the behavior is preferable, but I'm not sure we want to use that syntax due to unfamiliarity.